rse-bash.Rmd

```diff
@@ -0,0 +1,3902 @@
+standard output is connected to a terminal.  The LS_COLORS environment
+variable can change the settings.  Use the dircolors command to set it.
+
+Exit status:
+ 0  if OK,
+ 1  if minor problems (e.g., cannot access subdirectory),
+ 2  if serious trouble (e.g., cannot access command-line argument).
+
+GNU coreutils online help: <http://www.gnu.org/software/coreutils/>
+Full documentation at: <http://www.gnu.org/software/coreutils/ls>
+or available locally via: info '(coreutils) ls invocation'
+```
+
+> **Unsupported command-line options**
+>
+> If you try to use an option (flag) that is not supported, `ls` and other commands
+> will usually print an error message similar to:
+>
+> ```shell
+> $ ls -j
+> ```
+>
+> ```text
+> ls: invalid option -- 'j'
+> Try 'ls --help' for more information.
+> ```
+
+#### The `man` command
+
+The other way to learn about `ls` is to type
+```shell
+$ man ls
+```
+
+This will turn your terminal into a page with a description
+of the `ls` command and its options and, if you're lucky, some examples
+of how to use it.
+
+To navigate through the `man` pages,
+you may use <kbd>↑</kbd> and <kbd>↓</kbd> to move line-by-line,
+or try <kbd>B</kbd> and <kbd>Spacebar</kbd> to skip up and down by a full page.
+To search for a character or word in the `man` pages,
+use <kbd>/</kbd> followed by the character or word you are searching for.
+Sometimes a search will result in multiple hits.  If so, you can move between hits using <kbd>N</kbd> (for moving forward) and <kbd>Shift</kbd>+<kbd>N</kbd> (for moving backward).
+
+To **quit** the `man` pages, press <kbd>Q</kbd>.
+
+> **Manual pages on the web**
+>
+> Of course there is a third way to access help for commands:
+> searching the internet via your web browser.
+> When using internet search, including the phrase `unix man page` in your search
+> query will help to find relevant results.
+>
+> GNU provides links to its
+> [manuals](http://www.gnu.org/manual/manual.html) including the
+> [core GNU utilities](http://www.gnu.org/software/coreutils/manual/coreutils.html),
+> which covers many commands introduced within this lesson.
+
+
+### Looking around and moving about
+
+We can also use `ls` to see the contents of a different directory.  Let's take a
+look at our `Desktop` directory by running `ls -F Desktop`,
+i.e.,
+the command `ls` with the `-F` **option** and the **argument**  `Desktop`.
+The argument `Desktop` tells `ls` that
+we want a listing of something other than our current working directory:
+
+```shell
+$ ls -F Desktop
+```
+
+```text
+data-shell/
+```
+
+Your output should be a list of all the files and sub-directories on your
+Desktop, including the `data-shell` directory you downloaded at
+the [setup for this lesson](http://swcarpentry.github.io/shell-novice/setup.html).
+Take a look at your Desktop to confirm that your output is accurate.
+
+As you may now see, using a bash shell is strongly dependent on the idea that
```

Comment:
I don't see how using bash is strong dependent on hierarchical organization...?

Reply:
I've removed that whole paragraph. I don't think we need to talk about hierarchical organization.